os/kernel/fault_manager/fault_manager.c

```diff
@@ -0,0 +1,285 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <tinyara/init.h>
+#include <tinyara/board.h>
+#include <debug.h>
+#include <fcntl.h>
+#include "fault_manager.h"
+#ifdef CONFIG_BOARDCTL_RESET
+#include <sys/boardctl.h>
+#include <stdlib.h>
+#endif
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Global Data
+ ****************************************************************************/
+
+/* This structure is used to store the mapping between app id and application path
+ * Its value is filled at a time of registering the application for fault manager
+ */
+struct apptable reg_table[CONFIG_NUM_APPS] = { { -1, " "}, { -1, " "} };
+
+/****************************************************************************
+ * Name: fm_register_app
+ *
+ * Description:
+ *   This function is used to register application for the fault manager.
+ *   This function will map application pid with its application path.
+ *
+ * Input parameters:
+ *   pid    - The pid of the task to be registered
+ *   result - Application path to be registered for the fault manager
+ *
+ * Returned Value:
+ *   Zero (OK) on success; otherwise a -1 (ERROR)  value is returned.
+ *
+ ****************************************************************************/
+int fm_register_app(pid_t pid, const char *path)
+{
+	int i;
+
+	for (i = 0; i < CONFIG_NUM_APPS; i++) {
+		if (!(strcmp(reg_table[i].app_path, path))) {
+			reg_table[i].appid = pid;
+			return OK;
+		}
+	}
+
+	for (i = 0; i < CONFIG_NUM_APPS; i++) {
+		if (reg_table[i].appid == -1) {
+			reg_table[i].appid = pid;
+			strcpy(reg_table[i].app_path, path);
+			return OK;
+			lldbg("ERROR=%d OK =%d\n", ERROR, OK);
+		}
+	}
+
+	svdbg("Total number of apps exceeded\n");
```

Comment:
please define and use debug macro for fault manager, not svdbg or something.

Reply:
Added macro for fault manager